config.dev.toml

```diff
@@ -85,6 +85,17 @@ Port = ":3000"
 # that contains the letter `D` (for "Direct Access") in the first line.
 GlobalEndpoint = "http://localhost:3001"
 
+# Enables webhooks that allow you to two things outside of the standard 
+# module download protocol:
+#
+#   - Trigger Athens to download a module asynchronously, without downloading
+#       any module code or metadata yourself
+#   - Tell Athens that a vanity module path is backed by a given VCS path.
+#       When you do this, Athens will download at the version in the /@latest
+#       endpoint for the given VCS path, store it, and register the given VCS
+#       path with the vanity domain forever
+EnableWebhooks = false
```

Comment:
This wouldn't work unless we're supporting GitHub webhooks (and not GitLab, BitBucket etc) 
So we might want to just make an enum where "empty" means off.

Reply:
@marwan-at-work I don't understand. If you turned on webhooks, couldn't you htt them with any tool? They're essentially `curl` commands

Reply:
@arschles yes, but we can't control what the json shape of the body request will look like. We need to unmarshal the body to know what the module/version/event is...unless we make the url smart enough to figure all of that out which might be a weird devex

Reply:
Oh, well we need to specify what the JSON needs to be. I don't think we should necessarily use the GH webhook format, but we could. I didn't specify it in here yet, so here's an idea:

"Async fetch" webhook:

```console
GET /download/async/github.com/my/module
```

"Synchronous alias" webhook:

```console
POST /alias/sync/alias.com

{"origin": "github.com/my/module"}
```

Thoughts?

Reply:
If we don't use the GH webhook format, how will we know what version just got released from that webhook call?

Reply:
We have the option to handle all vcs webhook formats (github/gitlab/bitbucket/etc) or we could start with a library and cmd for integration into other tooling like `goreleaser`.  I think the latter would be a better starting point, then we can consider a webhook-detection routine or separate endpoints for handling the webhooks from each vcs site.

Reply:
@marwan-at-work @Kunde21 I think the version should be included in the async fetch webhook - probably as `.../@v/v1.0.1` at the end of the path above - but we can make it optional in the sync alias webhook.

Overall, I'm not looking for any support for any GH webhook formats in here (or other VCS's). IO want to define a simple format that someone can `curl` against to tell Athens either about the existence of a new module path & version, or about a new alias.

Reply:
@arschles that sitll doesn't work as a webhook, because the version is a moving target. We can't just have one version to be overridden